Snippets/Core/Core_6/Msmq/DLQOptInForTTBR.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace Core6.Msmq
+{
+    using NServiceBus;
+
+    class DLQOptInForTTBR
+    {
+        DLQOptInForTTBR(EndpointConfiguration endpointConfiguration)
+        {
+            #region msmq-dlq-for-ttbr-optin
+
+            endpointConfiguration.UseTransport<MsmqTransport>()
+                .UseDeadLetterQueueForMessagesWithTimeToReachQueue();
```

Comment:
>According to this article, there are 2 ways that messages can reach DLQ. When the TTBR or TTRQ timers elapse. This API talks about TTRQ. But the heartbeat etc, don't we set the TTBR?

TTRQ is defaulted to infinite by us. Users can override this using the `timeToReachQueue` flag in the [connection string](https://docs.particular.net/nservicebus/msmq/connection-strings)

TTBR is the thing that users can control with `TimeToBeReceived` attribute. 

>This is all about not DLQing based on TTBR not TTRQ, yet the API is called TTRQ. IMO, that's incorrect.

I agree, will update the core API to `UseDeadLetterQueueForMessagesWithTimeToBeReceived` cc @Particular/nservicebus-maintainers